addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,199 @@
+        if self.tax_id.l10n_ar_withholding_amount_type == 'untaxed_amount':
+            base_amount = factor * sum(tax_base_lines.mapped('price_subtotal'))
+        else:
+            base_amount = factor * sum(tax_base_lines.mapped('price_total'))
+        return self.payment_register_id.currency_id.round(base_amount / conversion_rate)
+
+
+    def _tax_compute_all_helper(self):
+        self.ensure_one()
+        # Computes the withholding tax amount provided a base and a tax
+        # It is equivalent to: amount = self.base * self.tax_id.amount / 100
+        taxes_res = self.tax_id.compute_all(
+            self.base_amount,
+            currency=self.payment_register_id.currency_id,
+            quantity=1.0,
+            product=False,
+            partner=False,
+            is_refund=False,
+        )
+        tax_amount = taxes_res['taxes'][0]['amount']
+        tax_account_id = taxes_res['taxes'][0]['account_id']
+        tax_repartition_line_id = taxes_res['taxes'][0]['tax_repartition_line_id']
+        return tax_amount, tax_account_id, tax_repartition_line_id
+
+    @api.depends('tax_id', 'base_amount')
+    def _compute_amount(self):
+        for line in self:
+            if not line.tax_id:
+                line.amount = 0.0
+            else:
+                line.amount, dummy, dummy = line._tax_compute_all_helper()
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    withholding_ids = fields.One2many(
+        'account.payment.register.withholding', 'payment_register_id', string="Withholdings",
+        compute='_compute_withholdings', readonly=False, store=True)
+    net_amount = fields.Monetary(compute='_compute_net_amount', readonly=True,  help="Net amount after withholdings")
+
+    @api.depends('line_ids', 'can_group_payments', 'group_payment')
+    def _compute_withholdings(self):
+        supplier_recs = self.filtered(lambda x: x.partner_type == 'supplier' and (not x.can_group_payments or (x.can_group_payments and x.group_payment)))
+        for rec in supplier_recs:
+            taxes = rec._get_withholding_tax()
+            rec.withholding_ids = [Command.clear()] + [Command.create({'tax_id': x.id, 'base_amount': 0}) for x in taxes]
+        (self - supplier_recs).withholding_ids = False
+
+    @api.depends('withholding_ids.amount', 'amount', 'l10n_latam_check_id')
+    def _compute_net_amount(self):
+        for rec in self:
+            if rec.l10n_latam_check_id:
+                rec.currency_id = rec.l10n_latam_check_id.currency_id
+                rec.net_amount = rec.l10n_latam_check_id.amount
+                net_info = rec._get_amount_net_info()
+                witholding_net_amount = min(rec.l10n_latam_check_id.amount , net_info[1])
+                rec.amount = witholding_net_amount * net_info[0] + (rec.l10n_latam_check_id.amount - witholding_net_amount)
```

Comment:
The amount should not have been set in another way (because this is the compute for net amount)

We could inherit the _compute_amount when l10n_latam_check_id is set.  Or inheriting https://github.com/odoo/odoo/blob/a34e5d20c86f42351d9582342dc928ec21117c7b/addons/l10n_latam_check/wizards/account_payment_register.py#L44 which has an onchange for it (but why did not we do a compute?)

Reply:
updated from this pr https://github.com/odoo/odoo/pull/130305/files#diff-a006380282c121a102360b6b57d8f0d82a3d6c63e9e400992b0427838b8784ef